### ISSUE TITLE
Metadata generation failed

### ISSUE THREAD

**Issue by letzfets** [Azure/azure-functions-durable-js#334]
**Describe the bug**
Trying to run the sample project

**Investigative information**

- Durable Functions extension version: 1.8.3
- durable-functions npm module version: ??
- Language (JavaScript/TypeScript) and version: -> sample
- Node.js version: v16.14.2
- dotnet version: 6.0.103

**To Reproduce**

1. func extensions install -p Microsoft.Azure.WebJobs.Extensions.DurableTask -v 1.8.32.

**Expected behavior**
install durabel functions extensions

**Actual behavior**
crashes with
````
...dataGenerator.targets(20,5): error : Metadata generation failed. [/<PATH-TO_REPO>/azure-functions-durable-js/samples/extensions.csproj]
````

**Comment by cgillum** [comment]
It looks like the samples in this repo are incredibly out of date. I recommend instead following the instructions here for getting started: https://docs.microsoft.com/en-us/azure/azure-functions/durable/quickstart-js-vscode

@davidmrdavid FYI, we should look into updating all these samples. It looks like they haven't been touched in 2-3 years.

**Comment by letzfets** [comment]
True. The samples in [azure-functions-durable-extension](https://github.com/Azure/azure-functions-durable-extension/tree/dev/samples) seem a bit newer. But the issue tracker for _javascript_ refers to this repo.

And that one creates the same error anyways.

I'm trying to run the samples with **function core tools** from command line (without VS Code): do you have a link to a tutorial / sample / documentation for that as well? I know how to run functions - just want to extend my knowledge to durable functions.

**Comment by cgillum** [comment]
I see. I suspect the reason for the crash is that the extension version being referenced is really old or doesn't exist (not sure if there's a typo in your command above). The latest extension version is **2.6.1**.

However, in most cases this step shouldn't be needed. Durable Functions can be used with extension bundles, removing the need for an extensions.csproj file or the `func extensions install` command.

I couldn't find instructions for how to do this with just Core Tools alone, but I tried a few things locally on my box and put together some instructions. Give this a try:

```bash
# Create a new function app
func init --worker-runtime node --language javascript

# Add the core Durable Functions functions
func new --name "DurableFunctionsHttpStart" --template "Durable Functions HTTP starter"
func new --name "DurableFunctionsOrchestrator" --template "Durable Functions orchestrator"
func new --name "Hello" --template "Durable Functions activity"

# Install the Durable Functions SDK
npm install durable-functions
```

You'll also want to make sure your local.settings.json points to a valid storage account for `AzureWebJobsStorage`. By default it tries to use a local emulator. If you don't have a local emulator handly, you can use the following docker command to run Azurite locally:

```bash
docker run --name 'azurite' -p 10000:10000 -p 10001:10001 -p 10002:10002 -d "mcr.microsoft.com/azure-storage/azurite:3.15.0"
```

To start the functions host, run:

```
func host start
```

To test the app to make sure everything's working, run this:

```
POST http://localhost:7071/api/orchestrators/DurableFunctionsOrchestrator
```

If you get a JSON response back with some URLs in it, then things should be working. :)

**Comment by davidmrdavid** [comment]
Thanks all for the discussion so far. @letzfets feel free to ping me if Chris' instructions above don't work for you. Indeed, I'm already tracking an update of this samples as in particular. I'll look to follow-up on it shortly

**Comment by letzfets** [comment]
Negativ @cgillum & @davidmrdavid :

Now I get the same error after running
````
func new --name "DurableFunctionsHttpStart" --template "Durable Functions HTTP starter"
````

I.e. it crashes with 
````
...dataGenerator.targets(20,5): error : Metadata generation failed. [/<PATH-TO_REPO>/azure-functions-durable-js/samples/extensions.csproj]
````

I've realised this is a rather inactive repository - but I believe it's still pretty relevant. Unless you can point me to anything newer for _durable functions with JavaScript_

Cheers 🍻

**Comment by cgillum** [comment]
Given that there is an extensions.csproj file mentioned in the error message, I feel like you might be going down a path that's not expected.

Are you running these steps in a new folder? If not, please try creating a new folder outside of this repro and running the steps I provided. It looks like you might be trying to run them in the existing samples folder, which might be causing unexpected behavior due to certain preexisting files being present.

**Comment by davidmrdavid** [comment]
Hi @lefzfets,

I just tried reproducing this both a new project (in a new directory), as well as in the samples directory of this project and I wasn't able to reproduce this behavior. Admittedly, in the `samples` directory I do get a few warnings due to some old files that need updating, but no exceptions.

Do you mind attaching a `.zipped` reproducer for this behavior? I can use that to help understand what's going on.

That said, installing the durable functions extension should be unnecessary unless you're doing some advanced scenario. In general, the steps for quickly scaffolding a Durable Functions app for NodeJS are those shown [here](https://docs.microsoft.com/en-us/azure/azure-functions/durable/quickstart-js-vscode). Do these work for you? Is there a reason why you need to install the Durable Functions Extension using the `func` CLI?

Finally, I also wonder if you somehow have an old version of the `func` CLI installed. Can you tell me what the output is for `func --v`? For me, it's `4.0.4426`. Thanks!

_I'll also look to update the samples right now, although from my testing just now those are not the cause of this error_

**Comment by davidmrdavid** [comment]
Just to be clear, the instructions @cgillum provided should be used to initialize a **new** Durable Functions project in a **new** folder. If they're executed inside the `samples` directory of this repo, they might not work as expected since the `samples` directory already contain the files that are meant to be generated by cgillum's command.

In general, the `samples` directory in this repo is currently best thought-of as a container of code examples, not as a runnable directory; I think this is part of where the confusion might be coming from. Admittedly, this is something we should fix on our end.

As I mentioned above, please do try [these](https://docs.microsoft.com/en-us/azure/azure-functions/durable/quickstart-js-vscode) instructions to scaffold your Durable Functions for NodeJS app; those should definitely work. Currently, utilizing `func` to do the equivalent setup is a bit of an advanced scenario in my opinion.

We're happy to help guide you through these steps as well, so please do reach out and we'll be glad to get you all set up :) .

**Comment by letzfets** [comment]
> I see. I suspect the reason for the crash is that the extension version being referenced is really old or doesn't exist (not sure if there's a typo in your command above). The latest extension version is **2.6.1**.
> 
> However, in most cases this step shouldn't be needed. Durable Functions can be used with extension bundles, removing the need for an extensions.csproj file or the `func extensions install` command.
> 
> I couldn't find instructions for how to do this with just Core Tools alone, but I tried a few things locally on my box and put together some instructions. Give this a try:
> 
> ```shell
> # Create a new function app
> func init --worker-runtime node --language javascript
> 
> # Add the core Durable Functions functions
> func new --name "DurableFunctionsHttpStart" --template "Durable Functions HTTP starter"
> func new --name "DurableFunctionsOrchestrator" --template "Durable Functions orchestrator"
> func new --name "Hello" --template "Durable Functions activity"
> 
> # Install the Durable Functions SDK
> npm install durable-functions
> ```
> 
> You'll also want to make sure your local.settings.json points to a valid storage account for `AzureWebJobsStorage`. By default it tries to use a local emulator. If you don't have a local emulator handly, you can use the following docker command to run Azurite locally:
> 
> ```shell
> docker run --name 'azurite' -p 10000:10000 -p 10001:10001 -p 10002:10002 -d "mcr.microsoft.com/azure-storage/azurite:3.15.0"
> ```
> 
> To start the functions host, run:
> 
> ```
> func host start
> ```
> 
> To test the app to make sure everything's working, run this:
> 
> ```
> POST http://localhost:7071/api/orchestrators/DurableFunctionsOrchestrator
> ```
> 
> If you get a JSON response back with some URLs in it, then things should be working. :)

**Comment by letzfets** [comment]
@cgillum : Yes - your suggestion above works!
@davidmrdavid : yes - my bad. I haven't realised, that I need to run it in a new directory.
My version `func -v` is `4.0.3971`.

For people trying to replicate the working suggestion from @cgillum above and running the function **locally**: You need to make sure, that you have a _storage account_ configured:
**TL;DR**: add a `connection string` to a storage account in `local.settings.json`

**full documentation**:
- with storage in Azure (tested): [Code and test Azure Functions locally](https://docs.microsoft.com/en-us/azure/azure-functions/functions-develop-local#local-settings-file)
- with local test storage (untested): [Use the Azurite emulator for local Azure Storage development](https://docs.microsoft.com/en-us/azure/storage/common/storage-use-azurite?tabs=npm)

I'm pretty sure @cgillum intention with the docker command is to use _Azurite_ but running the commands above does not link the functions to the azurite docker.
So other than updating the sample code, we might improve this patch by linking the docker container with the function? 😉

In any case: thanks for your great support @cgillum & @davidmrdavid 🙏

**Comment by davidmrdavid** [comment]
So glad to hear this issue got resolved, @lefzfets. Do reach out again if you need any more assistance :-) .

And yes, we'll be following up on the sample-folder updates.
@hossam-nasr: Just a heads up - let's coordinate to update the `samples` directory to make it easily runnable. It currently contains mixed PLs (TS and JS) which prevents us from actually running `func host start` in it.

**Comment by hossam-nasr** [comment]
Closing in favor of #356 and #357
